unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/RPCSocialApiBridge.cs

```diff
@@ -0,0 +1,310 @@
+using Cysharp.Threading.Tasks;
+using DCL;
+using DCl.Social.Friends;
+using DCL.Social.Friends;
+using Decentraland.Renderer.RendererServices;
+using Decentraland.Social.Friendships;
+using Google.Protobuf.WellKnownTypes;
+using JetBrains.Annotations;
+using RPC;
+using System;
+using System.Threading;
+using rpc_csharp;
+using rpc_csharp.transport;
+using RPC.Transports;
+using System.Collections.Generic;
+using UnityEngine;
+using Payload = Decentraland.Social.Friendships.Payload;
+
+namespace MainScripts.DCL.Controllers.FriendsController
+{
+    public class RPCSocialApiBridge : IRPCSocialApiBridge
+    {
+        private const int REQUEST_TIMEOUT = 30;
+
+        private string accessToken;
+
+        private readonly IUserProfileBridge userProfileWebInterfaceBridge;
+
+        public event Action<UserStatus> OnFriendAdded;
+        public event Action<string> OnFriendRemoved;
+        public event Action<FriendRequest> OnFriendRequestAdded;
+        public event Action<string> OnFriendRequestRemoved;
+        public event Action<AddFriendRequestsPayload> OnFriendRequestsAdded;
+        public event Action<FriendshipUpdateStatusMessage> OnFriendshipStatusUpdated;
+        public event Action<FriendRequestPayload> OnFriendRequestReceived;
+
+        private ClientFriendshipsService socialClient;
+
+        public RPCSocialApiBridge(MatrixInitializationBridge matrixInitializationBridge, IUserProfileBridge userProfileWebInterfaceBridge)
+        {
+            matrixInitializationBridge.OnReceiveMatrixAccessToken += (token) => { this.accessToken = token; };
+            this.userProfileWebInterfaceBridge = userProfileWebInterfaceBridge;
+        }
+
+        public async UniTaskVoid InitializeClient(CancellationToken cancellationToken = default)
+        {
+            var transport = new WebSocketClientTransport("wss://rpc-social-service.decentraland.zone");
+            var client = new RpcClient(transport);
+            var socialPort = await client.CreatePort("social-service-port");
+            var module = await socialPort.LoadModule(FriendshipsServiceCodeGen.ServiceName);
+            socialClient = new ClientFriendshipsService(module);
+
+            InitializeSubscriptions(cancellationToken).Forget();
+        }
+
+        public async UniTaskVoid InitializeSubscriptions(CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            await UniTask.WaitUntil(() =>
```

Comment:
Can we rethink `WaitUntil`? It looks ugly and not very efficient as it creates a delegate with every invocation